src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -114,39 +100,28 @@ public void PopulateItemGroups(PatternSet definition, IList<ContentItemGroup> co
             {
                 var groupPatterns = definition.GroupExpressions;
 
-                List<Tuple<ContentItem, Asset>> groupAssets = null;
+                List<(ContentItem Item, Asset Asset)> groupAssets = null;
+
                 foreach (var asset in _assets)
                 {
                     foreach (var groupPattern in groupPatterns)
                     {
                         var item = groupPattern.Match(asset.Path, definition.PropertyDefinitions);
                         if (item != null)
                         {
-                            if (groupAssets == null)
-                            {
-                                groupAssets = new List<Tuple<ContentItem, Asset>>(1);
-                            }
-
-                            groupAssets.Add(Tuple.Create(item, asset));
+                            groupAssets ??= new List<(ContentItem Item, Asset Asset)>(capacity: 1);
```

Comment:
I see that the capacity of the list is set to `1` by default and there is no `break` statement in this loop. Hence, I assume that the collection may have to resized. Is there any way we can estimate the default capacity to avoid the resizing in the next iteration?

Reply:
The size could be anywhere between zero and `_assets.Count * groupPatterns.Count`. I can't think of a good way to estimate without walking the loops twice, which would be a net-negative I'm sure.

Reply:
The capacity also used to be 1, so I'm assuming there might've been some intent behind that.